modules/programs/git.nix

```diff
@@ -415,12 +428,28 @@ in {
       (filterAttrs hasSmtp config.accounts.email.accounts);
     }
 
-    (mkIf (cfg.signing != null) {
-      programs.git.iniContent = {
-        user.signingKey = mkIf (cfg.signing.key != null) cfg.signing.key;
-        commit.gpgSign = mkDefault cfg.signing.signByDefault;
-        tag.gpgSign = mkDefault cfg.signing.signByDefault;
-        gpg.program = cfg.signing.gpgPath;
+    (mkIf (cfg.signing != { }) {
+      programs.git = {
+        signing = {
+          format = mkIf (versionOlder config.home.stateVersion "24.11")
+            (mkOptionDefault "openpgp");
+          signer = mkIf (cfg.signing.format != null) (mkOptionDefault {
+            openpgp = getExe' pkgs.gnupg "gpg2";
```

Comment:
nitpick
```suggestion
            openpgp = getExe pkgs.gnupg;
```

Reply:
This was from my PR which was made before `meta.mainProgram` was set for GnuPG :p

Reply:
I think we can just keep this for symmetry with gpgsm :D

Reply:
Anyway, now implemented.